Include/pymacro.h

```diff
@@ -23,6 +23,45 @@
 #  define static_assert _Static_assert
 #endif
 
+
+// _Py_ALIGN_AS: this compiler's spelling of `alignas` keyword,
+// We currently use alignas for free-threaded builds only; additional compat
+// checking would be great before we add it to the default build.
+// Standards/compiler support:
+// - `alignas` is a keyword in C23 and C++11.
+// - `_Alignas` is a keyword in C11
+// - GCC & clang has __attribute__((aligned))
+//   (use that for older standards in pedantic mode)
+// - MSVC has __declspec(align)
+// - `_Alignas` is common C compiler extension
+// Older compilers may name it differently; to allow compilation on such
+// unsupported platforms, we don't redefine _Py_ALIGN_AS if it's already
+// defined. Note that defining it wrong (including defining it to nothing) will
+// cause ABI incompatibilities.
+#ifdef Py_GIL_DISABLED
+#   ifndef _Py_ALIGN_AS
+#       ifdef __cplusplus
+#           if (__cplusplus < 201103L) \
+                  && (defined(__GNUC__) || defined(__clang__))
+#               define _Py_ALIGN_AS(V) __attribute__((aligned(V)))
+#           else
+#               define _Py_ALIGN_AS(V) alignas(V)
```

Comment:
This will be used if `__cplusplus < 201103L` and neither `__GNUC__` nor `__clang__` are defined.

Reply:
That's unknown compilers that aren't in C++11+-standard mode.
IMO, the best we can do is hope that they do support this part of C++11.

Reply:
What if it is an old Microsoft compiler? Then we could use `__declspec(align(V))`.